### ISSUE TITLE
SCRAM-SHA-256(-PLUS) supports

### ISSUE THREAD

**Issue by Neustradamus** [Jajcus/pyxmpp2#95]
"When using the SASL SCRAM mechanism, the SCRAM-SHA-256-PLUS variant SHOULD be preferred over the SCRAM-SHA-256 variant, and SHA-256 variants [RFC7677] SHOULD be preferred over SHA-1 variants [RFC5802]".

After:
- SCRAM-SHA-1(-PLUS), currently there is only SCRAM-SHA-1:
-- RFC5802: Salted Challenge Response Authentication Mechanism (SCRAM) SASL and GSS-API Mechanisms: https://tools.ietf.org/html/rfc5802
-- RFC6120: Extensible Messaging and Presence Protocol (XMPP): Core: https://tools.ietf.org/html/rfc6120

Can you add support for?
- SCRAM-SHA-256(-PLUS):
-- RFC7677: SCRAM-SHA-256 and SCRAM-SHA-256-PLUS Simple Authentication and Security Layer (SASL) Mechanisms: https://tools.ietf.org/html/rfc7677 - since 2015-11-02
-- RFC8600: Using Extensible Messaging and Presence Protocol (XMPP) for Security Information Exchange: https://tools.ietf.org/html/rfc8600 - since 2019-06-21: https://mailarchive.ietf.org/arch/msg/ietf-announce/suJMmeMhuAOmGn_PJYgX5Vm8lNA

In more, I add too;
- SCRAM-SHA-224(-PLUS)
- SCRAM-SHA-384(-PLUS)
- SCRAM-SHA-512(-PLUS): https://xmpp.org/extensions/inbox/hash-recommendations.html

--------------------------------------------------------------------------------------------------
-PLUS variants:
- RFC5056: On the Use of Channel Bindings to Secure Channels: https://tools.ietf.org/html/rfc5056
- RFC5929: Channel Bindings for TLS: https://tools.ietf.org/html/rfc5929
- Channel-Binding Types: https://www.iana.org/assignments/channel-binding-types/channel-binding-types.xhtml

LDAP:
- RFC5803: Lightweight Directory Access Protocol (LDAP) Schema for Storing Salted: Challenge Response Authentication Mechanism (SCRAM) Secrets: https://tools.ietf.org/html/rfc5803

HTTP:
- RFC7804: Salted Challenge Response HTTP Authentication Mechanism: https://tools.ietf.org/html/rfc7804

IANA:
- Simple Authentication and Security Layer (SASL) Mechanisms: https://www.iana.org/assignments/sasl-mechanisms/sasl-mechanisms.xhtml

Linked to:
- https://github.com/scram-xmpp/info/issues/1

**Comment by Neustradamus** [comment]
@Jajcus: Can you look it?

And add SCRAM-SHA-512/SCRAM-SHA-512-PLUS and SCRAM-SHA3-512/SCRAM-SHA3-512-PLUS in the same time?

Thanks in advance.

**Comment by Neustradamus** [comment]
@ProgVal: I think that you can help here? :)

**Comment by progval** [comment]
Not really, no. I'm not familiar with -PLUS variants and haven't worked with pyxmpp2 for a while.

**Comment by Jajcus** [comment]
@Neustradamus  it hurts to admit, but this project is dead now :-(

I won't be able to get back to it and bring it to current Python and XMPP standard. I am too much detached from this code now and don't have time to get back in.

But I do admire and appreciate your persistence and trust in the project. Issue from 2019, then comment in 2022 and now. I wish I could help.
